packages/salesforcedx-vscode-core/src/commands/templates/forceApexClassCreate.ts

```diff
@@ -39,36 +38,22 @@ export class ForceApexClassCreateExecutor extends BaseTemplateCommand {
       .withLogName('force_apex_class_create')
       .build();
   }
-
-  public sourcePathStrategy: SourcePathStrategy = PathStrategyFactory.createDefaultStrategy();
-
-  public getDefaultDirectory() {
-    return APEX_CLASS_DIRECTORY;
-  }
-
-  public getFileExtension() {
-    return APEX_CLASS_EXTENSION;
-  }
 }
 
 const fileNameGatherer = new SelectFileName();
 const outputDirGatherer = new SelectOutputDir(APEX_CLASS_DIRECTORY);
+const metadataTypeGatherer = new MetadataTypeGatherer(APEX_CLASS_TYPE);
 
 export async function forceApexClassCreate() {
   const commandlet = new SfdxCommandlet(
     new SfdxWorkspaceChecker(),
     new CompositeParametersGatherer<DirFileNameSelection>(
```

Comment:
`DirFileNameSelection` can be changed to `LocalComponent` now that the gatherers are building `{ type, fileName, outputDir }`